pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -440,7 +447,24 @@ def save_checkpoint(self, filepath, weights_only: bool = False) -> None:
         _checkpoint = self.dump_checkpoint(weights_only)
         self.trainer.accelerator.save_checkpoint(_checkpoint, filepath)
 
-    def _get_loops_state_dict(self):
+    def _get_lightning_module_state_dict(self) -> Dict[str, torch.Tensor]:
+        if _fault_tolerant_enabled():
+            metrics = [m for m in self.trainer.lightning_module.modules() if isinstance(m, Metric)]
+            for metric in metrics:
+                metric.persistent(True)
+                if not metric._is_synced:
+                    metric.sync()
```

Comment:
Already checked inside

```suggestion
                metric.sync()
```

Reply:
This is required actually. Metric API applies a hard sync and this would break on cpu.

Reply:
True, my bad.

I thought this `sync` call was for

```python
    def sync(self) -> None:
        for result_metric in self.result_metrics:
            if result_metric.is_tensor and not result_metric._is_synced:
                result_metric.sync()
```

Reply:
Done !